torch/_C/__init__.pyi.in

```diff
@@ -118,6 +118,14 @@ class _LegacyVariableBase(object):
     ) -> None: ...
 
 # Defined in torch/csrc/jit/python/init.cpp
+class Future(object):
+  def __init(self) -> None: ...
+  def wait(self) -> Any: ...
+  def then(self, callback: Callable) -> Future: ...
+  def set_result(self, result: Any) -> None: ...
```

Comment:
I wonder if it's a good idea to declare it as Generic[T]
```suggestion
class Future(Generic[T]):
  def __init(self) -> None: ...
  def wait(self) -> T: ...
  def then(self, callback: Callable[T]) -> Future[T]: ...
  def set_result(self, result: T) -> None: ...
```

Reply:
This looks good to me for `wait()` and `set_result()`. But `then()` might take a Callable that returns a different type.

cc @xush6528

Reply:
`Callable[[T], S]) -> Future[S]`